### PR TITLE
Add markdown response format for MCP tools

### PR DESCRIPTION
**Pull request by evantahler**
## Summary

- Adds `responseFormat` to `McpActionConfig` so actions can opt into markdown responses instead of JSON
- Every MCP tool gets a `_responseFormat` parameter that agents can use to override the action default per-call
- New `toMarkdown()` serializer renders objects as bulleted lists, uniform arrays as tables, and falls back to JSON code blocks beyond a configurable depth limit (`MCP_MARKDOWN_DEPTH_LIMIT`, default 5)
- Updates docs: MCP guide (new Response Format section), agents guide (best practices bullet), config guide (new config key)

## Test plan

- [x] 22 unit tests for `toMarkdown()` covering flat objects, nested objects, arrays, tables, depth limits, edge cases
- [x] 48 MCP integration tests including new tests for `_responseFormat` parameter and markdown rendering
- [x] Manual verification with an MCP client requesting markdown responses

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: example/backend/__tests__/initializers/mcp.test.ts [hunk 1]
```diff
@@ -261,6 +261,17 @@ describe("mcp initializer (enabled)", () => {
...
+    test("tool input schema does not include _responseFormat parameter", async () => {
+      const result = await client.listTools();
+      const statusTool = result.tools.find((t) => t.name === "status");
+      expect(statusTool).toBeDefined();
+
+      const schema = statusTool!.inputSchema as {
+        properties?: Record<string, any>;
+      };
+      expect(schema.properties?._responseFormat).toBeUndefined();
+    });
```
Comment (evantahler) [inline example/backend/__tests__/initializers/mcp.test.ts]: This test doesn't need to exist
